src/main/webapp/app/shared/form/title-channel-name/title-channel-name.component.ts

```
@@ -98,22 +85,31 @@ export class TitleChannelNameComponent implements AfterViewInit, OnDestroy, OnIn
-    formatChannelName(newName: string, allowDuplicateHyphens = true, removeTrailingHyphens = false) {
+    /**
+     * Formats a channel name by applying specific transformations based on the provided options.
+     *
+     * @param {string} newName - The new channel name to be formatted.
+     * @param {boolean} [allowDuplicateHyphens=true] - Flag indicating whether duplicate hyphens should be allowed in the formatted name.
+     * @param {boolean} [removeTrailingHyphens=false] - Flag indicating whether trailing hyphens should be removed from the formatted name.
+     * @return {void} This method does not return a value but emits the formatted channel name.
+     */
+    formatChannelName(newName: string, allowDuplicateHyphens: boolean = true, removeTrailingHyphens: boolean = false): void {
         const specialCharacters: RegExp = allowDuplicateHyphens ? /[^a-z0-9-]+/g : /[^a-z0-9]+/g;
         const trailingHyphens = removeTrailingHyphens ? /-$/ : new RegExp('[]');
```

Biome (1.9.4)

[error] Use a regular expression literal instead of the RegExp constructor.

Regular expression literals avoid some escaping required in a string literal, and are easier to analyze statically.
Safe fix: Use a literal notation instead.


(lint/complexity/useRegexLiterals)